src/Homepage/EntriesGroup.js

```diff
@@ -23,10 +21,27 @@ function Overlay(props) {
 function Timestamp({ Timestamp: value }) {
   const then = new Date(value);
   const now = new Date();
+  const timeDifference = now - then;
+  const HOURS_24 = 24 * 60 * 60 * 1000;
+  const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+
+  let hh = then.getHours();
+  let mm = then.getMinutes();
+  let period = (hh < 12) ? "am" : "pm";
+  hh = (hh === 0 || hh === 12) ? 12 : (hh % 12);
+  mm = (mm < 10) ? ("0" + mm) : mm;
+
+  let time = hh + ":" + mm + " " + period;
+
+  if (timeDifference >= HOURS_24 && timeDifference < (7 * HOURS_24)) {
+    time += " on " + days[then.getDay()];
+  }
+  else if (timeDifference >= (7 * HOURS_24)) {
+    time += " " + (timeDifference / HOURS_24).toFixed() + " days ago";
+  }
+
```

Comment:
The website preview looks good, though the code can be made bit more readable. We have integrated [date-fns](https://date-fns.org/docs/Getting-Started) package in this project. It has some nice utility functions. Do you think any of those can be used here ?

Reply:
Sure, I'll take a look at it.